llvm/include/llvm/CodeGen/AsmPrinter.h

```diff
@@ -20,6 +20,7 @@
 #include "llvm/ADT/SmallVector.h"
 #include "llvm/BinaryFormat/Dwarf.h"
 #include "llvm/CodeGen/AsmPrinterHandler.h"
+#include "llvm/CodeGen/DebugHandlerBase.h"
```

Comment:
Can this be avoided somehow? It pulls the entire kitchen sink into AsmPrinter.

Reply:
Right now, I don't think so, HandlerInfo is templated and unique_ptr requires the definition for the destructor. After #97046, which removes HandlerInfo, it should be possible with an out-of-line destructor of AsmPrinter. I'll check next week (I won't merge before then anyway) -- thanks for pointing this out.